loglevels-log4j2/src/main/scala/akka/management/loglevels/log4j2/LogLevelRoutes.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2017-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.management.loglevels.log4j2
+
+import akka.actor.{ ExtendedActorSystem, Extension, ExtensionId }
+import akka.annotation.InternalApi
+import akka.event.{ Logging => ClassicLogging }
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.unmarshalling.Unmarshaller
+import akka.management.scaladsl.{ ManagementRouteProvider, ManagementRouteProviderSettings }
+import org.apache.logging.log4j.{ Level, LogManager }
+import org.apache.logging.log4j.core.LoggerContext
+import org.slf4j.LoggerFactory
+
+object LogLevelRoutes extends ExtensionId[LogLevelRoutes] {
+
+  override def createExtension(system: ExtendedActorSystem): LogLevelRoutes =
+    new LogLevelRoutes(system)
+
+}
+
+/**
+ * Provides the path loglevel/logger which can be used to dynamically change log levels
+ *
+ * INTERNAL API
+ */
+@InternalApi
+final class LogLevelRoutes private (system: ExtendedActorSystem) extends Extension with ManagementRouteProvider {
+
+  private val log = LoggerFactory.getLogger(classOf[LogLevelRoutes])
+
+  import LoggingUnmarshallers._
+
+  override def routes(settings: ManagementRouteProviderSettings): Route = {
+    pathPrefix("loglevel") {
+      extractClientIP { clientIp =>
+        path("log4j2") {
+          pathEndOrSingleSlash {
+            put {
+              parameters(
+                "level".as[Level].withDefault(Level.INFO),
+                "logger" ? LogManager.ROOT_LOGGER_NAME
+              ) { (level, logger) =>
+                if (settings.readOnly) {
+                  complete(StatusCodes.Forbidden)
+                } else {
+                  log.info(
+                    s"Log level for [${if (logger.isBlank) "Root" else logger}] set to [$level] through Akka Management loglevel endpoint from [$clientIp]")
```

Comment:
`isBlank` is not available on jdk8, which we currently still support

Reply:
Fixed.